staging/src/k8s.io/cloud-provider/app/controllermanager.go

```diff
@@ -162,6 +168,97 @@ the cloud specific control loops shipped with Kubernetes.`,
 	return cmd
 }
 
+func createOrUpdateWebhookConfiguration(ctx context.Context, webhooks map[string]WebhookHandler, webhooksConfig cloudproviderconfig.WebhookConfiguration, clientBuilder clientbuilder.SimpleControllerClientBuilder) error {
```

Comment:
This self-registration of webhooks is *really* tricky to get right... I had no idea this was being contemplated to be added into cloud-controller-manager

some quick observations:
1. it's not clear why having CCM create this programatically is better than applying a manifest
2. does having CCM do this mean that multiple CCM controllers duel when the configuration changes?
3. is the WebhookAddress expected to point to the CCM instance? how does routing work when there are multiple CCM instances?
4. how do multiple CCM controllers handle rollout of a new webhook / configuration? all instances have to understand the new endpoints or fail open when requests to unknown endpoints arrive

Reply:
great questions @liggitt , i was not aware of the complexity around the self-registration but it makes perfect sense once you've laid it out. i think this might cause us to reconsider the methodology here, it seems like using a manifest would make a better experience for users and help with some of the ordering/dueling issues.

thanks for the review!